Add support for Feather icons

kleiram opened the pull request:
This PR adds support for the Feather Icons icon set and resolves issue #538.

It also, apparently, fixed some whitespace issues in the README 😄.

kleiram commented:
Note: I'm having some issues with the font family name at the moment, so it's not actually ready to merge just yet!

oblador commented:
Hi @kleiram,

Thanks a bunch for your contribution! To be able to upgrade when new icons are released I have a script in the `package.json` that generates these files (usually from `.svg` or `.css` files). Would you mind adding this?

Would prefer if you didn't commit all the whitespace changes in the readme, the `.editorconfig` is explicit in allowing this for `.md` files. 

With regards to TypeScript definitions, they live in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-native-vector-icons), so please exclude them and make a PR there instead.

kleiram commented:
@oblador I will create a new PR without the whitespace changes and TypeScript definitions. To be honest, the TS definitions weren't supposed to be there! 

About the script, I'm not 100% on how to use it. I created the icon set using icomoon. I guess I should add Feather Icons as a dev dependency, no?

oblador commented:
Yep, exactly; add `feather-icons` as devDependency and then use `fontcustom` to generate a font file & CSS, then the bundled `generate-icon` script to generate the rest of the files. Should be just copy pasting one of the other scripts and modifying inputs.

oblador commented:
Seems like you might run into problems, see [this comment](https://github.com/colebemis/feather/issues/135#issuecomment-315510347) for potential workaround.

kleiram commented:
Alright! I'll get to work on that! Thanks! 👍

kleiram commented:
@oblador I've reverted my previous commits and added support for Feather using the process you've described! Also, the TypeScript definitions are removed.

PS: About the TypeScript definitions. It is also possible to include a `index.d.ts` definition file to the root of the project (which TypeScript supports). Is it something you just don't want? Because if it is added to DefinitelyTyped, users have to install `@types/react-native-vector-icons`. When it is added in the root of the project, it is automatically installed. I'm willing to invest some time into that!

oblador commented:
Now it's perfect, thanks so much for your contribution!

kleiram commented:
Thanks for merging!

oblador commented:
Released it in [4.4.0](https://github.com/oblador/react-native-vector-icons/releases/tag/v4.4.0) if you haven't seen it already 😄

wcandillon commented:
I'm also excited to see to see 4.4.0.
